site/content/entry/yuzu-progress-report-feb-2021/index.md

```diff
@@ -0,0 +1,223 @@
+
+Additionally, he changed the implementation of [fibers](https://en.wikipedia.org/wiki/Fiber_(computer_science)) to [use unique_ptr instead of shared_ptr](https://github.com/yuzu-emu/yuzu/pull/6006), but later [changed it again to use weak_ptr](https://github.com/yuzu-emu/yuzu/pull/6041) as it's more appropiate in this use case.
+
+Fibers are similar to threads, except they can't be executed in parallel.
+Instead, they *yield* control to other fibers in a process.
+In yuzu, they're used to have better control over thread scheduling, working as tools for the kernel to quickly pause and resume emulated guest threads from within the application, without having to rely on the OS scheduler.
+
+Previously, fibers were managed through a special object called a `shared pointer`, a kind of variable that stores the memory addresses of other objects - it “references” them.
+In particular, this variable keeps track of how many `shared_ptr` references to an object exist in the program, and the memory won't be freed until the total amount of references is zero (i.e. when the object isn't being used anymore).
+If these references aren't managed with proper care, some pointers may retain memory and never free it, resulting in memory leaks.
+For this reason, bunnei changed the implementation to use a different kind of object, a `weak pointer`, which is similar to the `shared pointer` but it doesn't increase the reference counter, nor is it capable of deleting the original referenced object.
+Thus, the referenced memory will be free only when the original pointer is deleted, regardless of how many other `weak_ptr` references to the same memory exist, eliminating the memory leaks caused by the old implementation.
+
+One of the many tasks of the kernel is to assign resources to processes whenever they ask for them.
+For this reason, [epicboy](https://github.com/ameerj) started the work necessary to [utilize a more accurate resource_limit implementation](https://github.com/yuzu-emu/yuzu/pull/5877), in order to match the hardware behaviour more closely.
+
+Be it memory, threads, or ports, the kernel checks for their availability and keeps track of them through a variable called `resource limit`.
+By comparing the current amount of resources being used against the resource limit, the kernel can determine whether to deny a request or not.
+This stems from the fact that resources are finite, especially in weaker hardware such as that in the Nintendo Switch.
+A PC, on the other hand, isn't as restricted as a Nintendo Switch.
+Until now, whenever a process requested resources, yuzu would create its own instance of `resource limit` instead of using a system-wide variable to keep track of it.
+This PR is just the initial step in preparation to reverse engineer the correct behaviour and implement it in the emulator.
+
+## Paint me like one of your french bits
```

Comment:
It's intended to be funny more than accurate. I figured the reference would click on most people once they start reading about how the GPU can basically be used to operate over information (bits) even though it might not necessarily be textures or images. And yeah, I considered pixels, but really it's hard to think of something different from bit that fits as nicely here haha
Will let flam decide over this one.